### PR TITLE
[rpc] add limitation on GetBlocks and GetLogs. Also some other small changes

### PR DESCRIPTION
**Pull request by JackyWYX**
## Issue

Add limitation to GetBlocks and GetLogs.

### Unit Test Coverage

Passed local tests.

Passed mainnet RPC tests.

```
curl -s --request POST 'http://localhost:9500/' --header 'Content-Type: application/json' --data-raw '{ "jsonrpc": "2.0", "method": "hmyv2_getBlocks", "params": [1, 1025, {"withSigners": false, "fullTx": false,"inclStaking": false}], "id": 1}'
> .... # Some data

curl -s --request POST 'http://localhost:9500/' --header 'Content-Type: application/json' --data-raw '{ "jsonrpc": "2.0", "method": "hmyv2_getBlocks", "params": [1, 1026, {"withSigners": false, "fullTx": false,"inclStaking": false}], "id": 1}'
{"jsonrpc":"2.0","id":1,"error":{"code":-32000,"message":"GetBlocks query must be smaller than size 1024"}}
```

### CODE REVIEW

File: rpc/filters/filter.go [hunk 1]
```diff
@@ -220,6 +220,11 @@ func (f *Filter) unindexedLogs(ctx context.Context, end uint64) ([]*types.Log, e
 	var logs []*types.Log
 
 	for ; f.begin <= int64(end); f.begin++ {
+		select {
+		case <-ctx.Done():
```
Comment (rlan35) [inline rpc/filters/filter.go:224]: so we didn't have timeout on these ranging rpcs?

File: rpc/common/types.go [hunk 2]
```diff
@@ -13,9 +13,8 @@ import (
 // BlockArgs is struct to include optional block formatting params.
 type BlockArgs struct {
 	WithSigners bool     `json:"withSigners"`
-	InclTx      bool     `json:"inclTx"`
 	FullTx      bool     `json:"fullTx"`
-	Signers     []string `json:"signers"`
+	Signers     []string `json:"-"`
```
Comment (rlan35) [inline rpc/common/types.go:17]: why change to -?

File: rpc/blockchain.go [hunk 3]
```diff
@@ -188,7 +188,6 @@ func (s *PublicBlockchainService) GetBlockByNumber(
 			return nil, err
 		}
 	}
-	blockArgs.InclTx = true
```
Comment (rlan35) [inline rpc/blockchain.go:191]: was this a bug that all query got txn included in the return?

**Review by rlan35 — APPROVED** [review]

File: rpc/filters/filter.go (hunk 1 above)
Comment (JackyWYX) [reply rpc/filters/filter.go:224]: No. Even now we do not have timeout mechanism in this context. I would suggest to add a default timeout for all RPCs after Alex migrated Ethereum RPC module.

File: rpc/common/types.go (hunk 2 above)
Comment (JackyWYX) [reply rpc/common/types.go:17]: The field is just a field to store in-memory computation result, and is not used in RPC calls. Thus I disabled the json serialization of this field. 

Reference at https://api.hmny.io/#52f8a4ce-d357-46f1-83fd-d100989a8243.

File: rpc/blockchain.go (hunk 3 above)
Comment (JackyWYX) [reply rpc/blockchain.go:191]: This looks like a deprecated field and it is not used in any cases. Thus I removed this field.

**Pull request merged**